pkg/authorizer/token_retriever.go

```diff
@@ -12,63 +13,88 @@ import (
 	"time"
 
 	"github.com/Azure/msi-acrpull/pkg/authorizer/types"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
 )
 
 const (
 	defaultARMResource              = "https://management.azure.com/"
 	customARMResourceEnvVar         = "ARM_RESOURCE"
 	msiMetadataEndpoint             = "http://169.254.169.254/metadata/identity/oauth2/token"
 	defaultCacheExpirationInSeconds = 600
+	authorityHost                   = "https://login.microsoftonline.com/"
+	resource                        = "https://management.azure.com/.default"
```

Comment:
In workload identity token retriever, add `/.default` to `defaultARMResource`

Reply:
Done for next commit.